### PR TITLE
feat(hooks): Add useOnKeyDown hook

### PR DESCRIPTION
**Pull request by 0length**
### Overview 
This is refer to https://github.com/GetJobber/atlantis/pull/394#discussion_r509685744
### Changed

Create new hook called `useOnKeyDown` to create on keydown event listener handler by specific one or multyple key in the same function.
**Thanks**

### CODE REVIEW

File: packages/hooks/src/useOnKeyDown/useOnKeyDown.mdx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    const letterText2 = "Space Enter and Shift + Escape will show here";
+    const [ text, setText ] = useState(letterText);
+    const [ textTwo, setTextTwo ] = useState(letterText2);
+    useOnKeyDown(() => setText("Deleted. Press Control + z for undo."), "Delete");
+    useOnKeyDown(() => setText(letterText), {key: "z", ctrlKey: true});
+    useOnKeyDown((e) => {
+        if(e.shiftKey) setTextTwo("You pressed Shift and "+e.key);
+        else setTextTwo("You pressed '"+e.key+"'");
+    }, [" ", "Enter", {key: "Escape", shiftKey: true} ]);
+    return (<Text>{text}<pre />{textTwo}</Text>);}}
```
Comment (seanhealy) [inline packages/hooks/src/useOnKeyDown/useOnKeyDown.mdx:35]: Looks like we have an extra `<pre />` here.

File: packages/hooks/src/useOnKeyDown/useOnKeyDown.mdx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+`useOnKeyDown` should **only** only by used when making keyboard shortcut in any component.
+
+```tsx
+import { useOnKeyDown } from "@jobber/hooks";
+```
+
+
+<Playground>
+  {()=>{
+    const letterText = "Delete me by pressing Delete key on your keyboard.";
```
Comment (seanhealy) [inline packages/hooks/src/useOnKeyDown/useOnKeyDown.mdx:25]: Macs label backspace and delete quite oddly which make this example a bit confusing for users on that platform. We can probably get a way with simplifying this just down to a single example. How would you feel about removing this delete one and switching your second one to `escape` `space-enter` and something like `ctrl-i` or something?

File: packages/hooks/src/useOnKeyDown/useOnKeyDown.mdx (hunk 1 above)
Comment (seanhealy) [reply packages/hooks/src/useOnKeyDown/useOnKeyDown.mdx:35]: Might also be nice to keep the instructions fixed and have a second line that displays the result of the key events.

File: packages/hooks/src/useOnKeyDown/useOnKeyDown.mdx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import { Playground } from "docz";
+import { useState } from "react";
+import { useOnKeyDown } from "./useOnKeyDown";
+import { Text } from "@jobber/components/Text";
+
+# UseOnKeyDown
+
+`useOnKeyDown` is an extremly simple hook that can be for adding the keydown event handler when the component is mounted and remove them when it unmount.
+
+`useOnKeyDown` should **only** only by used when making keyboard shortcut in any component.
```
Comment (seanhealy) [inline packages/hooks/src/useOnKeyDown/useOnKeyDown.mdx:16]: Another little wording thought. 😄 

```suggestion
`useOnKeyDown` should **only ** be used when making keyboard shortcut in a component.
```

File: packages/hooks/src/useOnKeyDown/useOnKeyDown.mdx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+---
+
+import { Playground } from "docz";
+import { useState } from "react";
+import { useOnKeyDown } from "./useOnKeyDown";
+import { Text } from "@jobber/components/Text";
+
+# UseOnKeyDown
+
+`useOnKeyDown` is an extremly simple hook that can be for adding the keydown event handler when the component is mounted and remove them when it unmount.
```
Comment (seanhealy) [inline packages/hooks/src/useOnKeyDown/useOnKeyDown.mdx:14]: Little wording thought. 😄 

```suggestion
`useOnKeyDown` is a simple hook adding the keydown event handler when the component is mounted and removing when unmounted.
```

File: packages/hooks/src/useOnKeyDown/useOnKeyDown.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  readonly [index: string]: boolean | string | undefined;
+}
+
+type KeyComparerator = XOR<VerboseKeyComparerator, SimpleKeyComparerator>;
+
+export function useOnKeyDown(
+  callback: (event: globalThis.KeyboardEvent) => void,
+  keys: KeyComparerator[] | KeyComparerator,
+) {
+  const handler: (event: globalThis.KeyboardEvent) => void = (
```
Comment (seanhealy) [inline packages/hooks/src/useOnKeyDown/useOnKeyDown.ts:21]: Our conventions would prefer this function defined with the `function` keyword.

```suggestion
function handler(event: globalThis.KeyboardEvent) {
```

**Review by seanhealy — CHANGES_REQUESTED** [review]
Looking great! Thanks so much. 😄 

Had a few little thoughts. Other than those it would be nice to have a test for this hook and I think we should be able to now use this hook [here so we don't have a duplicate](https://github.com/GetJobber/atlantis/blob/master/packages/components/src/Autocomplete/Menu.tsx#L160-L174).

**Comment by chris-at-jobber** [comment]
@0length we've fixed an issue causing specs to fail - if you update this branch from the latest master specs will pass

**Comment by 0length** [comment]
> @0length we've fixed an issue causing specs to fail - if you update this branch from the latest master specs will pass

Y

**Comment by 0length** [comment]
> @0length we've fixed an issue causing specs to fail - if you update this branch from the latest master specs will pass

Yeah.. I have a problem with my leptop.. Maybe i will work the changes in 2-3day later.. And create with the test so..

**Comment by seanhealy** [comment]
@0length Sorry to hear that your laptop is having a problem. Thanks so much for all your help during Hacktoberfest though! I'll mark this with the accepted label and we can sort the rest out later on. 😃

**Comment by chris-at-jobber** [comment]
Closing - this got taken over the finish line in a [separate PR](https://github.com/GetJobber/atlantis/pull/432). Thanks for kicking this off @0length !

**Pull request closed without merging**